inginious/frontend/pages/admin/admin.py

```diff
@@ -23,15 +24,34 @@ def POST_AUTH(self, *args, **kwargs):
 
     def show_page(self):
         """Display page"""
-
-        page = int(request.form.get("page")) if request.form.get("page") is not None else 1
-        user_per_page = 10  # TODO probably better to let user define user_per_page
-        all_users = self.user_manager.get_users_info(usernames=None, limit=user_per_page, skip=(page-1)*user_per_page)
-        size_users = self.user_manager.get_users_count()
-        pages = size_users // user_per_page + (size_users % user_per_page > 0) if user_per_page > 0 else 1
+        user_input = request.form
+        if "displayed_selection" in user_input:
+            # Mostly happened when changing page
+            params = json.loads(user_input.get("displayed_selection", ""))
+        else:
+            params={"limit":10,"sort_by":"username","order":1}
+            for key in ["username", "realname", "email"]:
+                params[key] = user_input[key] if key in user_input else ""
+            if "activated" in user_input:
+                params["activated"] = user_input["activated"]
+            if "sort_by" in user_input and "order" in user_input:
+                params["sort_by"]=user_input["sort_by"]
+                params["order"]= int(user_input["order"])
+            if "limit" in user_input:
+                params["limit"] = int(user_input["limit"])
+        query = {}
+        for key in ["username","realname","email"]:
+            query[key]={'$regex': f".*{params[key]}.*"}
```

Comment:
Do not add useless constraint to the query. If there is no filter to be applied, do not apply a dummy regex that will run uselessly.